### ISSUE TITLE
Can't build reactive-banana-wx

### ISSUE THREAD

**Issue by DevJac** [commercialhaskell/stack#2269]
Steps to reproduce:

1. `stack new test simple`
2. `cd` to the new `test` directory
3. Add `reactive-banana-wx` to your `build-depends` in `test.cabal`
4. Run `stack solver --update-config`
5. Add the following to your `stack.yaml` file (as described in issue #1316):
    
    ```
    explicit-setup-deps:
        "reactive-banana-wx": true
    ```
6. Run `stack build` in the `test` director created in step 1.

When I do this, I get the following output and error:
```
$ stack build
pqueue-1.3.1.1: configure
pqueue-1.3.1.1: build
cabal-macosx-0.2.3.4: configure
cabal-macosx-0.2.3.4: build
wxdirect-0.92.2.0: configure
wxdirect-0.92.2.0: build
cabal-macosx-0.2.3.4: copy/register
pqueue-1.3.1.1: copy/register
reactive-banana-1.1.0.1: configure
reactive-banana-1.1.0.1: build
wxdirect-0.92.2.0: copy/register
wxc-0.92.2.0: configure
reactive-banana-1.1.0.1: copy/register
wxc-0.92.2.0: build
wxc-0.92.2.0: copy/register
wxcore-0.92.2.0: configure
wxcore-0.92.2.0: build
wxcore-0.92.2.0: copy/register
wx-0.92.2.0: configure
wx-0.92.2.0: build
wx-0.92.2.0: copy/register
reactive-banana-wx-1.1.1.0: configure
Progress: 8/9
--  While building package reactive-banana-wx-1.1.1.0 using:
      /opt/ghc/bin/ghc --make -odir /tmp/stack29239/reactive-banana-wx-1.1.1.0/.stack-work/dist/x86_64-linux/Cabal-1.22.5.0/setup -hidir /tmp/stack29239/reactive-banana-wx-1.1.1.0/.stack-work/dist/x86_64-linux/Cabal-1.22.5.0/setup -i -i. -clear-package-db -global-package-db -package-db=/home/buttons/.stack/snapshots/x86_64-linux/lts-6.3/7.10.3/pkgdb -package-db=/home/buttons/Code/haskell/test/.stack-work/install/x86_64-linux/lts-6.3/7.10.3/pkgdb -hide-all-packages -package=Cabal-1.22.5.0 -package-id=array-0.5.1.0-960bf9ae8875cc30355e086f8853a049 -package-id=base-4.8.2.0-0d6d1084fbc041e1cded9228e80e264d -package-id=bin-package-db-0.0.0.0-c2998655922e5892be4d23c583233dd1 -package-id=binary-0.7.5.0-bf6a937522bb8cec5d00218470f28789 -package-id=bytestring-0.10.6.0-9a873bcf33d6ce2fd2698ce69e2c1c66 -package-id=cabal-macosx-0.2.3.4-50d7f4029fe9c435d6e77827398c06b6 -package-id=containers-0.5.6.2-59326c33e30ec8f6afd574cbac625bbb -package-id=deepseq-1.4.1.1-614b63b36dd6e29d2b35afff57c25311 -package-id=directory-1.2.2.0-660a7a83a753ed85c8a374c15dae2b97 -package-id=filepath-1.4.0.0-f97d1e4aebfd7a03be6980454fe31d6e -package-id=ghc-prim-0.4.0.0-6cdc86811872333585fa98756aa7c51e -package-id=haskeline-0.7.2.1-ee66b80ab55ef13718feae46e4940f5d -package-id=hoopl-3.10.0.2-e1d2f66d9c173061362d378e84471a99 -package-id=hpc-0.6.0.2-a33eebda3a4ae222891fd5d14c4f2b1f -package-id=integer-gmp-1.0.0.0-3c8c40657a9870f5c33be17496806d8d -package-id=pretty-1.1.2.0-5cc412214ea63f61ee84c4fbabdbe0ec -package-id=process-1.2.3.0-a22328103774f0499a990653944cbf99 -package-id=reactive-banana-1.1.0.1-f678e9a834e435b19d945d73add6f16c -package-id=template-haskell-2.10.0.0-3c4cb52230f347282af9b2817f013181 -package-id=terminfo-0.4.0.1-ff5b3d5838955b7711e16ccc720f06f4 -package-id=time-1.5.0.1-1b9a502bb07a3e6f4d6935fbf9db7181 -package-id=transformers-0.4.2.0-81450cd8f86b36eaa8fa0cbaf6efc3a3 -package-id=unix-2.7.1.0-bb54ee8f9f6b2600aae7a748eb88a610 -package-id=wx-0.92.2.0-03c6b2513f3f6ab311a0ac7a5581a030 -package-id=wxcore-0.92.2.0-c73ad66b4af460fef6e92397799131ae -package-id=xhtml-3000.2.1-e005917157f780654d68110616d034e8 /tmp/stack29239/reactive-banana-wx-1.1.1.0/Setup.hs -o /tmp/stack29239/reactive-banana-wx-1.1.1.0/.stack-work/dist/x86_64-linux/Cabal-1.22.5.0/setup/setup
    Process exited with code: ExitFailure 1
    Logs have been written to: /home/buttons/Code/haskell/test/.stack-work/logs/reactive-banana-wx-1.1.1.0.log

    [1 of 1] Compiling Main             ( /tmp/stack29239/reactive-banana-wx-1.1.1.0/Setup.hs, /tmp/stack29239/reactive-banana-wx-1.1.1.0/.stack-work/dist/x86_64-linux/Cabal-1.22.5.0/setup/Main.o )
    
    /tmp/stack29239/reactive-banana-wx-1.1.1.0/Setup.hs:11:19:
        Couldn't match type ‘LocalBuildInfo’
                       with ‘Cabal-1.22.8.0:Distribution.Simple.LocalBuildInfo.LocalBuildInfo’
        NB: ‘LocalBuildInfo’
              is defined in ‘Distribution.Simple.LocalBuildInfo’
                  in package ‘Cabal-1.22.5.0’
            ‘Cabal-1.22.8.0:Distribution.Simple.LocalBuildInfo.LocalBuildInfo’
              is defined in ‘Distribution.Simple.LocalBuildInfo’
                  in package ‘Cabal-1.22.8.0’
        Expected type: Args
                       -> Distribution.Simple.Setup.BuildFlags
                       -> Distribution.PackageDescription.PackageDescription
                       -> LocalBuildInfo
                       -> IO ()
          Actual type: [String]
                       -> Cabal-1.22.8.0:Distribution.Simple.Setup.BuildFlags
                       -> Cabal-1.22.8.0:Distribution.PackageDescription.PackageDescription
                       -> Cabal-1.22.8.0:Distribution.Simple.LocalBuildInfo.LocalBuildInfo
                       -> IO ()
        In the ‘postBuild’ field of a record
        In the second argument of ‘($)’, namely
          ‘simpleUserHooks {postBuild = myPostBuild}’
```

**Comment by chrisdone** [comment]
You have two different versions of Cabal installed:

```
        NB: ‘LocalBuildInfo’
              is defined in ‘Distribution.Simple.LocalBuildInfo’
                  in package ‘Cabal-1.22.5.0’
            ‘Cabal-1.22.8.0:Distribution.Simple.LocalBuildInfo.LocalBuildInfo’
              is defined in ‘Distribution.Simple.LocalBuildInfo’
                  in package ‘Cabal-1.22.8.0’
```

What's the output for the following?

```
stack exec which ghc
stack exec -- ghc-pkg list transformers
```

**Comment by DevJac** [comment]
```
~/Code/haskell/test $ stack exec which ghc
/opt/ghc/bin/ghc
~/Code/haskell/test $ stack exec -- ghc-pkg list transformers
/opt/ghc/lib/ghc-7.10.3/package.conf.d
   transformers-0.4.2.0
/home/buttons/.stack/snapshots/x86_64-linux/lts-6.3/7.10.3/pkgdb
/home/buttons/Code/haskell/test/.stack-work/install/x86_64-linux/lts-6.3/7.10.3/pkgdb
```
I know of 2 or 3 cabal installations on my system.
cabal 1.16 is installed from the Ubuntu repositories.
cabal 1.24 is installed in `~/.cabal/bin`; I installed this version by doing `cabal install cabal-install`
I'm guessing stack has it's own cabal version as well. I don't know where cabal 1.22.5 and 1.22.8 are installed; those are the ones I'm apparently having trouble with.

**Comment by chrisdone** [comment]
Sorry, I meant the output for `Cabal`, not transformers.

**Comment by DevJac** [comment]
No problem, thanks for your help. Output below:
```
~/Code/haskell/test $ stack exec -- ghc-pkg list Cabal
/opt/ghc/lib/ghc-7.10.3/package.conf.d
   Cabal-1.22.5.0
/home/buttons/.stack/snapshots/x86_64-linux/lts-6.3/7.10.3/pkgdb
   Cabal-1.22.8.0
/home/buttons/Code/haskell/test/.stack-work/install/x86_64-linux/lts-6.3/7.10.3/pkgdb
```

**Comment by chrisdone** [comment]
Oh, I see. `Setup.hs` depends on two different versions of Cabal?

**Comment by DeathByTape** [comment]
The _bad news_: I encountered this issue today and-- as a result-- I stumbled upon this unresolved issue.

The _good news_: it appears as though I've found a solution.

In short, it looks like, perhaps, something had changed in Cabal. The workaround is to simply add `Cabal-1.22.5.0` to your `extra-deps`. My `stack.yaml` looks as follows:

```yaml
# Specifies the GHC version and set of packages available (e.g., lts-3.5, nightly-2015-09-21, ghc-7.10.2)
resolver: lts-5.0

# Local packages, usually specified by relative directory name
packages:
- '.'

# Packages to be pulled from upstream that are not in the resolver (e.g., acme-missiles-0.3)
extra-deps:
  - Cabal-1.22.5.0
  - cabal-macosx-0.2.3.4
  - reactive-banana-1.1.0.1
  - reactive-banana-wx-1.1.1.0
  - wx-0.92.2.0
  - wxcore-0.92.2.0
  - wxc-0.92.2.0
  - wxdirect-0.92.2.0

explicit-setup-deps:
  "reactive-banana-wx": true
```

Now we have a happy reactive 🍌  setup. Good luck!

**Comment by mgsloan** [comment]
This issue should be fixed in stack >= 1.1.2 

https://github.com/commercialhaskell/stack/commit/0463159b67717f336762dcf7a07709089cb9a149

Are you on an older version?

**Comment by DeathByTape** [comment]
@mgsloan good catch. I'm on 1.0.2 :x I don't update my work laptop slack often enough. 👍

**Comment by DevJac** [comment]
I am using stack `1.1.2`:

```
$ stack --version
Version 1.1.2, Git revision cebe10e845fed4420b6224d97dcabf20477bbd4b (3646 commits) x86_64 hpack-0.14.0
```

**Comment by DevJac** [comment]
I think this issue got closed by mistake. Apparently this issue is suppose to be fixed in the latest version of stack, but I have the latest version of stack and am still having this issue (I created this ticket). I have been using stack `1.1.2` since the beginning of the beginning of this issue.

**Comment by mgsloan** [comment]
@DevJac Perhaps it is a cabal regression, then.  Can you please paste the `-v` output?  That way we can see how stack is building Setup.hs

**Comment by mgsloan** [comment]
@DevJac Oh, sorry, I thought @DeathByTape opened the issue.

NVM I see it is in your original log:

>      /opt/ghc/bin/ghc --make -odir /tmp/stack29239/reactive-banana-wx-1.1.1.0/.stack-work/dist/x86_64-linux/Cabal-1.22.5.0/setup -hidir /tmp/stack29239/reactive-banana-wx-1.1.1.0/.stack-work/dist/x86_64-linux/Cabal-1.22.5.0/setup -i -i. -clear-package-db -global-package-db -package-db=/home/buttons/.stack/snapshots/x86_64-linux/lts-6.3/7.10.3/pkgdb -package-db=/home/buttons/Code/haskell/test/.stack-work/install/x86_64-linux/lts-6.3/7.10.3/pkgdb -hide-all-packages -package=Cabal-1.22.5.0 -package-id=array-0.5.1.0-960bf9ae8875cc30355e086f8853a049 -package-id=base-4.8.2.0-0d6d1084fbc041e1cded9228e80e264d -package-id=bin-package-db-0.0.0.0-c2998655922e5892be4d23c583233dd1 -package-id=binary-0.7.5.0-bf6a937522bb8cec5d00218470f28789 -package-id=bytestring-0.10.6.0-9a873bcf33d6ce2fd2698ce69e2c1c66 -package-id=cabal-macosx-0.2.3.4-50d7f4029fe9c435d6e77827398c06b6 -package-id=containers-0.5.6.2-59326c33e30ec8f6afd574cbac625bbb -package-id=deepseq-1.4.1.1-614b63b36dd6e29d2b35afff57c25311 -package-id=directory-1.2.2.0-660a7a83a753ed85c8a374c15dae2b97 -package-id=filepath-1.4.0.0-f97d1e4aebfd7a03be6980454fe31d6e -package-id=ghc-prim-0.4.0.0-6cdc86811872333585fa98756aa7c51e -package-id=haskeline-0.7.2.1-ee66b80ab55ef13718feae46e4940f5d -package-id=hoopl-3.10.0.2-e1d2f66d9c173061362d378e84471a99 -package-id=hpc-0.6.0.2-a33eebda3a4ae222891fd5d14c4f2b1f -package-id=integer-gmp-1.0.0.0-3c8c40657a9870f5c33be17496806d8d -package-id=pretty-1.1.2.0-5cc412214ea63f61ee84c4fbabdbe0ec -package-id=process-1.2.3.0-a22328103774f0499a990653944cbf99 -package-id=reactive-banana-1.1.0.1-f678e9a834e435b19d945d73add6f16c -package-id=template-haskell-2.10.0.0-3c4cb52230f347282af9b2817f013181 -package-id=terminfo-0.4.0.1-ff5b3d5838955b7711e16ccc720f06f4 -package-id=time-1.5.0.1-1b9a502bb07a3e6f4d6935fbf9db7181 -package-id=transformers-0.4.2.0-81450cd8f86b36eaa8fa0cbaf6efc3a3 -package-id=unix-2.7.1.0-bb54ee8f9f6b2600aae7a748eb88a610 -package-id=wx-0.92.2.0-03c6b2513f3f6ab311a0ac7a5581a030 -package-id=wxcore-0.92.2.0-c73ad66b4af460fef6e92397799131ae -package-id=xhtml-3000.2.1-e005917157f780654d68110616d034e8 /tmp/stack29239/reactive-banana-wx-1.1.1.0/Setup.hs -o /tmp/stack29239/reactive-banana-wx-1.1.1.0/.stack-work/dist/x86_64-linux/Cabal-1.22.5.0/setup/setup

Note there is indeed a single Cabal dep, `-package=Cabal-1.22.5.0`.  You are probably using a snapshot that has `Cabal-1.22.8.0`, and that's being pulled in as a dep via `cabal-macosx`.  This will indeed cause issues.  It's actually a fundamental problem with supporting varying the Cabal version - anything that depends on it will need to also use that version.  I've opened a discussion [here](https://github.com/commercialhaskell/stack/issues/2280).

Basically, `Setup.hs` files are a royal pain and we should really avoid doing things like this when possible.

**Comment by DevJac** [comment]
```
~/Code/haskell/test $ stack build -v
Version 1.1.2, Git revision cebe10e845fed4420b6224d97dcabf20477bbd4b (3646 commits) x86_64 hpack-0.14.0
2016-06-17 16:15:50.713070: [debug] Checking for project config at: /home/buttons/Code/haskell/test/stack.yaml @(stack_IZ6kIIshaMC0uV4dtbkAts:Stack.Config src/Stack/Config.hs:811:9)
2016-06-17 16:15:50.713284: [debug] Loading project config file stack.yaml @(stack_IZ6kIIshaMC0uV4dtbkAts:Stack.Config src/Stack/Config.hs:829:13)
2016-06-17 16:15:50.717424: [debug] Checking whether stack was built with libgmp4 @(stack_IZ6kIIshaMC0uV4dtbkAts:Stack.Config src/Stack/Config.hs:326:5)
2016-06-17 16:15:50.717643: [debug] Run process: ldd /home/buttons/.local/bin/stack @(stack_IZ6kIIshaMC0uV4dtbkAts:System.Process.Read src/System/Process/Read.hs:283:3)
2016-06-17 16:15:50.726147: [debug] Stack was not built with libgmp4 @(stack_IZ6kIIshaMC0uV4dtbkAts:Stack.Config src/Stack/Config.hs:330:14)
2016-06-17 16:15:50.726295: [debug] Trying to decode /home/buttons/.stack/build-plan-cache/x86_64-linux/lts-6.3.cache @(stack_IZ6kIIshaMC0uV4dtbkAts:Data.Binary.VersionTagged src/Data/Binary/VersionTagged.hs:55:5)
2016-06-17 16:15:50.740889: [debug] Success decoding /home/buttons/.stack/build-plan-cache/x86_64-linux/lts-6.3.cache @(stack_IZ6kIIshaMC0uV4dtbkAts:Data.Binary.VersionTagged src/Data/Binary/VersionTagged.hs:64:13)
2016-06-17 16:15:50.741050: [debug] Getting system compiler version @(stack_IZ6kIIshaMC0uV4dtbkAts:Stack.Setup src/Stack/Setup.hs:341:17)
2016-06-17 16:15:50.741165: [debug] Run process: ghc --info @(stack_IZ6kIIshaMC0uV4dtbkAts:System.Process.Read src/System/Process/Read.hs:283:3)
2016-06-17 16:15:50.772411: [debug] Asking GHC for its version @(stack_IZ6kIIshaMC0uV4dtbkAts:Stack.Setup.Installed src/Stack/Setup/Installed.hs:94:13)
2016-06-17 16:15:50.772541: [debug] Run process: ghc --numeric-version @(stack_IZ6kIIshaMC0uV4dtbkAts:System.Process.Read src/System/Process/Read.hs:283:3)
2016-06-17 16:15:50.796591: [debug] Getting Cabal package version @(stack_IZ6kIIshaMC0uV4dtbkAts:Stack.GhcPkg src/Stack/GhcPkg.hs:165:5)
2016-06-17 16:15:50.796760: [debug] Run process: ghc-pkg --no-user-package-db field --simple-output Cabal version @(stack_IZ6kIIshaMC0uV4dtbkAts:System.Process.Read src/System/Process/Read.hs:283:3)
2016-06-17 16:15:50.813573: [debug] Resolving package entries @(stack_IZ6kIIshaMC0uV4dtbkAts:Stack.Setup src/Stack/Setup.hs:221:5)
2016-06-17 16:15:50.814025: [debug] Getting global package database location @(stack_IZ6kIIshaMC0uV4dtbkAts:Stack.GhcPkg src/Stack/GhcPkg.hs:48:5)
2016-06-17 16:15:50.814117: [debug] Run process: ghc-pkg --no-user-package-db list --global @(stack_IZ6kIIshaMC0uV4dtbkAts:System.Process.Read src/System/Process/Read.hs:283:3)
2016-06-17 16:15:50.833707: [debug] Run process: ghc-pkg --global --no-user-package-db dump --expand-pkgroot @(stack_IZ6kIIshaMC0uV4dtbkAts:System.Process.Read src/System/Process/Read.hs:283:3)
2016-06-17 16:15:50.874521: [debug] Ignoring package haskeline due to wanting version 0.7.2.3 instead of 0.7.2.1 @(stack_IZ6kIIshaMC0uV4dtbkAts:Stack.Build.Installed src/Stack/Build/Installed.hs:189:5)
2016-06-17 16:15:50.874677: [debug] Ignoring package terminfo due to wanting version 0.4.0.2 instead of 0.4.0.1 @(stack_IZ6kIIshaMC0uV4dtbkAts:Stack.Build.Installed src/Stack/Build/Installed.hs:189:5)
2016-06-17 16:15:50.874788: [debug] Ignoring package Cabal due to wanting version 1.22.8.0 instead of 1.22.5.0 @(stack_IZ6kIIshaMC0uV4dtbkAts:Stack.Build.Installed src/Stack/Build/Installed.hs:189:5)
2016-06-17 16:15:50.875274: [debug] Run process: ghc-pkg --user --no-user-package-db --package-db /home/buttons/.stack/snapshots/x86_64-linux/lts-6.3/7.10.3/pkgdb dump --expand-pkgroot @(stack_IZ6kIIshaMC0uV4dtbkAts:System.Process.Read src/System/Process/Read.hs:283:3)
2016-06-17 16:15:50.922508: [debug] Run process: ghc-pkg --user --no-user-package-db --package-db /home/buttons/Code/haskell/test/.stack-work/install/x86_64-linux/lts-6.3/7.10.3/pkgdb dump --expand-pkgroot @(stack_IZ6kIIshaMC0uV4dtbkAts:System.Process.Read src/System/Process/Read.hs:283:3)
2016-06-17 16:15:50.943097: [debug] Trying to decode /home/buttons/.stack/indices/Hackage/00-index.cache @(stack_IZ6kIIshaMC0uV4dtbkAts:Data.Binary.VersionTagged src/Data/Binary/VersionTagged.hs:55:5)
2016-06-17 16:15:51.165304: [debug] Success decoding /home/buttons/.stack/indices/Hackage/00-index.cache @(stack_IZ6kIIshaMC0uV4dtbkAts:Data.Binary.VersionTagged src/Data/Binary/VersionTagged.hs:64:13)
2016-06-17 16:15:51.196405: [debug] Getting global package database location @(stack_IZ6kIIshaMC0uV4dtbkAts:Stack.GhcPkg src/Stack/GhcPkg.hs:48:5)
2016-06-17 16:15:51.196636: [debug] Run process: ghc-pkg --no-user-package-db list --global @(stack_IZ6kIIshaMC0uV4dtbkAts:System.Process.Read src/System/Process/Read.hs:283:3)
2016-06-17 16:15:51.220063: [info] reactive-banana-wx-1.1.1.0: configure @(stack_IZ6kIIshaMC0uV4dtbkAts:Stack.Build.Execute src/Stack/Build/Execute.hs:731:23)
2016-06-17 16:15:51.220498: [debug] Run process: /opt/ghc/bin/ghc --make -odir /tmp/stack3418/reactive-banana-wx-1.1.1.0/.stack-work/dist/x86_64-linux/Cabal-1.22.5.0/setup -hidir /tmp/stack3418/reactive-banana-wx-1.1.1.0/.stack-work/dist/x86_64-linux/Cabal-1.22.5.0/setup -i -i. -clear-package-db -global-package-db -package-db=/home/buttons/.stack/snapshots/x86_64-linux/lts-6.3/7.10.3/pkgdb -package-db=/home/buttons/Code/haskell/test/.stack-work/install/x86_64-linux/lts-6.3/7.10.3/pkgdb -hide-all-packages -package=Cabal-1.22.5.0 -package-id=array-0.5.1.0-960bf9ae8875cc30355e086f8853a049 -package-id=base-4.8.2.0-0d6d1084fbc041e1cded9228e80e264d -package-id=bin-package-db-0.0.0.0-c2998655922e5892be4d23c583233dd1 -package-id=binary-0.7.5.0-bf6a937522bb8cec5d00218470f28789 -package-id=bytestring-0.10.6.0-9a873bcf33d6ce2fd2698ce69e2c1c66 -package-id=cabal-macosx-0.2.3.4-50d7f4029fe9c435d6e77827398c06b6 -package-id=containers-0.5.6.2-59326c33e30ec8f6afd574cbac625bbb -package-id=deepseq-1.4.1.1-614b63b36dd6e29d2b35afff57c25311 -package-id=directory-1.2.2.0-660a7a83a753ed85c8a374c15dae2b97 -package-id=filepath-1.4.0.0-f97d1e4aebfd7a03be6980454fe31d6e -package-id=ghc-prim-0.4.0.0-6cdc86811872333585fa98756aa7c51e -package-id=haskeline-0.7.2.1-ee66b80ab55ef13718feae46e4940f5d -package-id=hoopl-3.10.0.2-e1d2f66d9c173061362d378e84471a99 -package-id=hpc-0.6.0.2-a33eebda3a4ae222891fd5d14c4f2b1f -package-id=integer-gmp-1.0.0.0-3c8c40657a9870f5c33be17496806d8d -package-id=pretty-1.1.2.0-5cc412214ea63f61ee84c4fbabdbe0ec -package-id=process-1.2.3.0-a22328103774f0499a990653944cbf99 -package-id=reactive-banana-1.1.0.1-f678e9a834e435b19d945d73add6f16c -package-id=template-haskell-2.10.0.0-3c4cb52230f347282af9b2817f013181 -package-id=terminfo-0.4.0.1-ff5b3d5838955b7711e16ccc720f06f4 -package-id=time-1.5.0.1-1b9a502bb07a3e6f4d6935fbf9db7181 -package-id=transformers-0.4.2.0-81450cd8f86b36eaa8fa0cbaf6efc3a3 -package-id=unix-2.7.1.0-bb54ee8f9f6b2600aae7a748eb88a610 -package-id=wx-0.92.2.0-03c6b2513f3f6ab311a0ac7a5581a030 -package-id=wxcore-0.92.2.0-c73ad66b4af460fef6e92397799131ae -package-id=xhtml-3000.2.1-e005917157f780654d68110616d034e8 /tmp/stack3418/reactive-banana-wx-1.1.1.0/Setup.hs -o /tmp/stack3418/reactive-banana-wx-1.1.1.0/.stack-work/dist/x86_64-linux/Cabal-1.22.5.0/setup/setup @(stack_IZ6kIIshaMC0uV4dtbkAts:System.Process.Read src/System/Process/Read.hs:301:3)
Progress: 1/2
--  While building package reactive-banana-wx-1.1.1.0 using:
      /opt/ghc/bin/ghc --make -odir /tmp/stack3418/reactive-banana-wx-1.1.1.0/.stack-work/dist/x86_64-linux/Cabal-1.22.5.0/setup -hidir /tmp/stack3418/reactive-banana-wx-1.1.1.0/.stack-work/dist/x86_64-linux/Cabal-1.22.5.0/setup -i -i. -clear-package-db -global-package-db -package-db=/home/buttons/.stack/snapshots/x86_64-linux/lts-6.3/7.10.3/pkgdb -package-db=/home/buttons/Code/haskell/test/.stack-work/install/x86_64-linux/lts-6.3/7.10.3/pkgdb -hide-all-packages -package=Cabal-1.22.5.0 -package-id=array-0.5.1.0-960bf9ae8875cc30355e086f8853a049 -package-id=base-4.8.2.0-0d6d1084fbc041e1cded9228e80e264d -package-id=bin-package-db-0.0.0.0-c2998655922e5892be4d23c583233dd1 -package-id=binary-0.7.5.0-bf6a937522bb8cec5d00218470f28789 -package-id=bytestring-0.10.6.0-9a873bcf33d6ce2fd2698ce69e2c1c66 -package-id=cabal-macosx-0.2.3.4-50d7f4029fe9c435d6e77827398c06b6 -package-id=containers-0.5.6.2-59326c33e30ec8f6afd574cbac625bbb -package-id=deepseq-1.4.1.1-614b63b36dd6e29d2b35afff57c25311 -package-id=directory-1.2.2.0-660a7a83a753ed85c8a374c15dae2b97 -package-id=filepath-1.4.0.0-f97d1e4aebfd7a03be6980454fe31d6e -package-id=ghc-prim-0.4.0.0-6cdc86811872333585fa98756aa7c51e -package-id=haskeline-0.7.2.1-ee66b80ab55ef13718feae46e4940f5d -package-id=hoopl-3.10.0.2-e1d2f66d9c173061362d378e84471a99 -package-id=hpc-0.6.0.2-a33eebda3a4ae222891fd5d14c4f2b1f -package-id=integer-gmp-1.0.0.0-3c8c40657a9870f5c33be17496806d8d -package-id=pretty-1.1.2.0-5cc412214ea63f61ee84c4fbabdbe0ec -package-id=process-1.2.3.0-a22328103774f0499a990653944cbf99 -package-id=reactive-banana-1.1.0.1-f678e9a834e435b19d945d73add6f16c -package-id=template-haskell-2.10.0.0-3c4cb52230f347282af9b2817f013181 -package-id=terminfo-0.4.0.1-ff5b3d5838955b7711e16ccc720f06f4 -package-id=time-1.5.0.1-1b9a502bb07a3e6f4d6935fbf9db7181 -package-id=transformers-0.4.2.0-81450cd8f86b36eaa8fa0cbaf6efc3a3 -package-id=unix-2.7.1.0-bb54ee8f9f6b2600aae7a748eb88a610 -package-id=wx-0.92.2.0-03c6b2513f3f6ab311a0ac7a5581a030 -package-id=wxcore-0.92.2.0-c73ad66b4af460fef6e92397799131ae -package-id=xhtml-3000.2.1-e005917157f780654d68110616d034e8 /tmp/stack3418/reactive-banana-wx-1.1.1.0/Setup.hs -o /tmp/stack3418/reactive-banana-wx-1.1.1.0/.stack-work/dist/x86_64-linux/Cabal-1.22.5.0/setup/setup
    Process exited with code: ExitFailure 1
    Logs have been written to: /home/buttons/Code/haskell/test/.stack-work/logs/reactive-banana-wx-1.1.1.0.log

    [1 of 1] Compiling Main             ( /tmp/stack3418/reactive-banana-wx-1.1.1.0/Setup.hs, /tmp/stack3418/reactive-banana-wx-1.1.1.0/.stack-work/dist/x86_64-linux/Cabal-1.22.5.0/setup/Main.o )
    
    /tmp/stack3418/reactive-banana-wx-1.1.1.0/Setup.hs:11:19:
        Couldn't match type ‘LocalBuildInfo’
                       with ‘Cabal-1.22.8.0:Distribution.Simple.LocalBuildInfo.LocalBuildInfo’
        NB: ‘LocalBuildInfo’
              is defined in ‘Distribution.Simple.LocalBuildInfo’
                  in package ‘Cabal-1.22.5.0’
            ‘Cabal-1.22.8.0:Distribution.Simple.LocalBuildInfo.LocalBuildInfo’
              is defined in ‘Distribution.Simple.LocalBuildInfo’
                  in package ‘Cabal-1.22.8.0’
        Expected type: Args
                       -> Distribution.Simple.Setup.BuildFlags
                       -> Distribution.PackageDescription.PackageDescription
                       -> LocalBuildInfo
                       -> IO ()
          Actual type: [String]
                       -> Cabal-1.22.8.0:Distribution.Simple.Setup.BuildFlags
                       -> Cabal-1.22.8.0:Distribution.PackageDescription.PackageDescription
                       -> Cabal-1.22.8.0:Distribution.Simple.LocalBuildInfo.LocalBuildInfo
                       -> IO ()
        In the ‘postBuild’ field of a record
        In the second argument of ‘($)’, namely
          ‘simpleUserHooks {postBuild = myPostBuild}’
```

**Comment by mgsloan** [comment]
@DevJac Thanks, I actually realized your original paste has the ghc call as it's in the error report.  That was sufficient to figure it out and open #2280
